src/neuromaps_prime/graph.py

```diff
@@ -502,3 +761,262 @@ def get_node_data(self, node_name: str) -> Node:
+        for u, v, key, edge_data in self.edges(data=True, keys=True):
+            if u != source_space or v != target_space or key != "surface_to_surface":
+                continue
+            edge: Edge = edge_data["data"]
+            for transform in edge.surface_transforms:
+                if (
+                    (density is None or transform.density == density)
+                    and (hemisphere is None or transform.hemisphere == hemisphere)
+                    and (
+                        resource_type is None
+                        or transform.resource_type == resource_type
+                    )
+                ):
+                    matching_transforms.append(transform)
+        return matching_transforms
+
+    def find_common_density(self, mid_space: str, target_space: str) -> str:
+        """Find a common density between source and source-to-target transformations.
+
+        This function is really needed to check if the surface_sphere_project_unproject
+        function can be performed between two transformations via an intermediate space.
+        If you are going from space A to space C via space B,
+        you need to find the highest common density between the
+        space B atlases and B->C transforms. If not the transformation is not possible.
+        """
+        atlases = self.search_surface_atlases(space=mid_space)
+        print(f"Resources found for transformation: {atlases}")
+
+        transforms = self.search_surface_transforms(
+            source_space=mid_space, target_space=target_space
+        )
+        print(f"Transforms found for transformation: {transforms}")
+
+        # Find highest common density between list of atlases and transforms
+        atlas_densities = {atlas.density for atlas in atlases}
+        transform_densities = {transform.density for transform in transforms}
+        common_densities = atlas_densities & transform_densities
+
+        if common_densities:
+            # If densities are strings like "32k", sort numerically
+            highest_common_density = max(common_densities, key=_get_density_key)
+            print(f"Highest common density: {highest_common_density}")
+        else:
+            print("No common density found between atlases and transforms.")
+
+        return highest_common_density
+
+    def find_highest_density(self, space: str) -> str:
+        """Find the highest density available for a given space in the graph."""
+        atlases = self.search_surface_atlases(space=space)
+        densities = {atlas.density for atlas in atlases}
+
+        if not densities:
+            raise ValueError(f"No atlases found for space '{space}'.")
+
+        highest_density = max(densities, key=_get_density_key)
+        print(f"Highest density for space '{space}': {highest_density}")
+        return highest_density
+
+    def surface_to_surface_transformer(
+        self,
+        transformer_type: str,
+        input_file: Path,
```

Comment:
Hah, this one threw me off. I thought it was asking them to provide sphere for some reason, but this makes sense.